dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java

```diff
@@ -584,6 +586,17 @@ public AuthenticatedUser createAuthenticatedUser(UserRecordIdentifier userRecord
         return authenticatedUser;
     }
 
+    public AuthenticatedUser createAuthenticatedUser(UserRecordIdentifier userRecordId,
```

Comment:
The argument list is kinda big, I would stick to max ~3 as described in Clean Code book. I would extract linked fields to "value object" .

Reply:
I'm not convinced that's the best solution here. Only `userDisplayInfo` and `preferredNotificationsLanguage` are fields from the page. But we don't want for the service to be aware of Page and there is little to none logical connection between these two parameters so wrapping them together would feel clunky at best.

We can discuss this on daily.